### PR TITLE
implement apiBody parsing into single page template

### PR DESCRIPTION
**Pull request by SWheeler17**
This is the single page template update for the PR: https://github.com/apidoc/apidoc-core/pull/104. It implements the params apiBody and apiQuery in the documentations' UI.

Example screenshots (with dummy data):
Screen Shot 2020-10-19 at 15 16 51
Screen Shot 2020-10-19 at 15 17 07

### CODE REVIEW

**Comment by SWheeler17** [comment]
@NicolasCARPi may you please review this PR?

**Comment by NicolasCARPi** [comment]
@SWheeler17 Thank you for your contribution. I'll review it when I can. Feel free to ping me if it takes too much time.

**Comment by SWheeler17** [comment]
@NicolasCARPi, pinging you to review the PR. Is there someone else I should also ask to review the PR?

File: template-single/index.html [hunk 1]
```diff
@@ -1500,7 +1621,46 @@ <h2><del>{{__ compare.key}}</del></h2>
...
+          case "object":
+            if (strBody[element.field] === undefined) {
+              strBody[element.field] = {};
+            }
+            break;
+          default:
+            strBody[element.field] = null;
+        }
+      });
+      console.log(strBody);
```
Comment (NicolasCARPi) [inline template-single/index.html:1661]: this line should be removed

**Review by NicolasCARPi — CHANGES_REQUESTED** [review]
Did you try to run the test suite?

2020-11-13-161022_1119x651_scrot

File: template-single/index.html (hunk 1 above)
Comment (Megapixel99) [reply template-single/index.html:1661]: Thanks, good catch.

**Comment by Megapixel99** [comment]
Tests pass for me. Since Github won't let me upload an image (saying something went wrong) here is the output:

```
apidoc@0.25.0 test /Volumes/Work/apidoc
> npm run jshint && mocha test/


> apidoc@0.25.0 jshint /Volumes/Work/apidoc
> jshint lib/ test/



  apiDoc full example with no config path
    ✓ should find latest example version
    ✓ should create example in tmp/ (391ms)
    ✓ should find created files
    ✓ created files should equal to fixtures

  apiDoc full example with config path for .json file
    ✓ should find latest example version
    ✓ should create example in tmp/ (358ms)
    ✓ should find created files
    ✓ created files should equal to fixtures

  apiDoc full example with config path for .js file
    ✓ should find latest example version
    ✓ should create example in tmp/ (333ms)
    ✓ should find created files
    ✓ created files should equal to fixtures

  apiDoc full example with config path for dir
    ✓ should find latest example version
    ✓ should create example in tmp/ (330ms)
    ✓ should find created files
    ✓ created files should equal to fixtures

  apiDoc custom markdown parser
    ✓ should render static text with custom markdown parser
    ✓ should create example in tmp/ (337ms)
    ✓ should create example in tmp/ with relative parser location (320ms)
    ✓ should create example in tmp/ with cwd parser location (310ms)
    ✓ should create example in tmp/ with absolute parser location (309ms)
    ✓ should find created files
    ✓ created files should have custom text

  apiDoc multiple folder input
    ✓ should create apidoc in /tmp/apidocmulti with multiple input folders (329ms)
    ✓ created files should equal to fixtures

  send sample request utils
    ✓ should handle nested fields
    ✓ should not handle nested fields when the specified type different to "Object"
    ✓ should not handle nested fields when an Object has a value
    ✓ should handle array of strings field
    ✓ should handle Object and array of objects field
    ✓ should handle nested fields and parse array and object
    ✓ should properly parse boolean and number fields on a parsed object
    ✓ should not affect boolean and number fields which cannot be parsed
    ✓ should convert path params to the accepted format


  34 passing (3s)
```

**Comment by SWheeler17** [comment]
Tests run fine for me

Github isn't letting me upload images so here is the output:

```
apidoc@0.25.0 test /Volumes/Work/apidoc
> npm run jshint && mocha test/


> apidoc@0.25.0 jshint /Volumes/Work/apidoc
> jshint lib/ test/



  apiDoc full example with no config path
    ✓ should find latest example version
    ✓ should create example in tmp/ (391ms)
    ✓ should find created files
    ✓ created files should equal to fixtures

  apiDoc full example with config path for .json file
    ✓ should find latest example version
    ✓ should create example in tmp/ (358ms)
    ✓ should find created files
    ✓ created files should equal to fixtures

  apiDoc full example with config path for .js file
    ✓ should find latest example version
    ✓ should create example in tmp/ (333ms)
    ✓ should find created files
    ✓ created files should equal to fixtures

  apiDoc full example with config path for dir
    ✓ should find latest example version
    ✓ should create example in tmp/ (330ms)
    ✓ should find created files
    ✓ created files should equal to fixtures

  apiDoc custom markdown parser
    ✓ should render static text with custom markdown parser
    ✓ should create example in tmp/ (337ms)
    ✓ should create example in tmp/ with relative parser location (320ms)
    ✓ should create example in tmp/ with cwd parser location (310ms)
    ✓ should create example in tmp/ with absolute parser location (309ms)
    ✓ should find created files
    ✓ created files should have custom text

  apiDoc multiple folder input
    ✓ should create apidoc in /tmp/apidocmulti with multiple input folders (329ms)
    ✓ created files should equal to fixtures

  send sample request utils
    ✓ should handle nested fields
    ✓ should not handle nested fields when the specified type different to "Object"
    ✓ should not handle nested fields when an Object has a value
    ✓ should handle array of strings field
    ✓ should handle Object and array of objects field
    ✓ should handle nested fields and parse array and object
    ✓ should properly parse boolean and number fields on a parsed object
    ✓ should not affect boolean and number fields which cannot be parsed
    ✓ should convert path params to the accepted format


  34 passing (3s)
```

**Comment by NicolasCARPi** [comment]
Yes it was on my side! (symlink to a non master branch of apidoc-core in node_modules)

File: template-single/index.html [hunk 2]
```diff
@@ -1500,7 +1621,45 @@ <h2><del>{{__ compare.key}}</del></h2>
     Handlebars.registerHelper('nl2br', function(text) {
         return _handlebarsNewlineToBreak(text);
     });
-
+    /**
+     *
```
Comment (NicolasCARPi) [inline template-single/index.html:1625]: it would be more useful to use this comment to explain what this helper does and why it was added

**Comment by tommy87** [comment]
@NicolasCARPi 
While checking the `build-example` action, i found the following warnings
```
{"message":"parser plugin 'apiquery' not found in block: 0 in file: example/example.js","level":"warn"}
{"message":"parser plugin 'apiquery' not found in block: 1 in file: example/example.js","level":"warn"}
{"message":"parser plugin 'apibody' not found in block: 1 in file: example/example.js","level":"warn"}
{"message":"parser plugin 'apiquery' not found in block: 3 in file: example/example.js","level":"warn"}
```

I think this warning occurs because i am using the `npm` version of apidoc-core in the action.
So wouldnt it be better if we are using the dev branch for testing, or mabye create two actions to validate both configutations?

And second question, should the CI job fail when a warning occurs in the build step?

File: template-single/index.html (hunk 2 above)
Comment (SWheeler17) [reply template-single/index.html]: I have implemented added a comment explaining the purpose of the function in the latest commit.

**Comment by NicolasCARPi** [comment]
> create two actions to validate both configutations?

Yes. master and dev.

> And second question, should the CI job fail when a warning occurs in the build step?

Definitely yes.

**Comment by SWheeler17** [comment]
Has the CI job been updated? If so, is there any reason this should not be merged?

**Comment by NicolasCARPi** [comment]
Only the travis-ci is failing so it can be merged in dev at least imho.

**Pull request merged**